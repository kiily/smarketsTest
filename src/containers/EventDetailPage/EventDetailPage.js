import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCaretRight, faCalendarDay } from '@fortawesome/free-solid-svg-icons';

import './EventDetailPage.scss';
import { createMinimalEvent } from '../../utils/helpers';
class EventDetailPage extends Component {

  render() {
    const eventId = this.props.match.params.id;
    const event = this.props.popularEvents.find(event => event.id === eventId);
    let minimalEvent = null;
    // e.g. MONDAY, APRIL 29, 8:00 PM
    const detailDateFormat = 'EEEE, LLLL d, h:mm aa';
    if (event) {
      minimalEvent = createMinimalEvent(event, detailDateFormat);
    }
    return (
      minimalEvent &&
        <div className="EventDetailPage-container">
          <ol className="breadcrumb-nav">
            <li className="breadcrumb">{minimalEvent.sport}</li>
            <li className="breadcrumb">
              <div className="caret-container">
                <FontAwesomeIcon icon={faCaretRight}/>
              </div>
              {minimalEvent.competition}
            </li>
          </ol>
          <div className="event-header">
            <div className="teams-container">
              <h1 className="competitor">
                <span className="competitor-text">{minimalEvent.teams[0]}</span>
              </h1>
              <h1 className="competitor">
                <span className="competitor-text">{minimalEvent.teams[1]}</span>
              </h1>
            </div>
            <div className="event-info-container">
              <div className="event-date">
                <div className="calendar-icon-container">
                  <FontAwesomeIcon icon={faCalendarDay}/>
                </div>
                <span className="event-date-text">{minimalEvent.startTime}</span>
              </div>
            </div>
          </div>
        </div>
    );
  }
}

EventDetailPage.propTypes = {
  popularEvents: PropTypes.array,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string
    })
  })
};

export default withRouter(EventDetailPage);
