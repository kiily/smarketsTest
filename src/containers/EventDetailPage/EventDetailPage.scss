@import '../../styles/variables.scss';

.EventDetailPage-container {
    .breadcrumb-nav {
        display: flex;
        color: $white;
        align-items: center;
        list-style-type: none;
        padding: 10px;
        margin-bottom: 5px;
        .breadcrumb {
            @include uppercase-text();
            display: flex;
            align-items: center;
            .caret-container {
                display: flex;
                align-items: center;
                justify-content: center;
                font-size: $font-l;
                border-radius: 2px;
                width: 1.125rem;
                height: 1.125rem;
                background: $caret-grey;
                margin: 0 0.5rem;
            }
        }
    }
    .event-header {
        border-top: 1px solid $border-color;
        margin: 0 10px;
        padding: 0.5rem;
        .teams-container {
            .competitor {
                color: $white;
                font-size: $font-xl;
            }
        }
        .event-info-container {
            .event-date {
                @include uppercase-text($font-m);
                display: flex;
                color: $text-grey;
                .calendar-icon-container {
                    margin-right: 5px;
                }
            }
        }
    }
}