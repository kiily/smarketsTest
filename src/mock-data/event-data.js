export const events = [
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-16T10:09:18.291988Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/championship/2019/04/30/millwall-fc-vs-bristol-city-fc',
    'hidden': false,
    'parent': {
      'name': 'Championship'
    },
    'id': '1925896',
    'inplay_enabled': true,
    'modified': '2019-04-16T10:09:18.291988Z',
    'name': 'Millwall vs. Bristol City',
    'parent_id': '939285',
    'short_name': 'MWL vs. BRC',
    'slug': 'millwall-fc-bristol-city-fc-2019-04-30',
    'special_rules': null,
    'start_date': '2019-04-30',
    'start_datetime': '2019-04-30T18:45:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-17T10:10:36.025237Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/championship/2019/05/01/swansea-vs-derby',
    'hidden': false,
    'parent': {
      'name': 'Championship'
    },
    'id': '1934371',
    'inplay_enabled': true,
    'modified': '2019-04-17T10:10:36.025237Z',
    'name': 'Swansea City vs. Derby County',
    'parent_id': '939285',
    'short_name': 'SWA vs. DER',
    'slug': 'swansea-derby-2019-05-01',
    'special_rules': null,
    'start_date': '2019-05-01',
    'start_datetime': '2019-05-01T18:45:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-18T10:11:56.786364Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/europe-uefa-champions-league/2019/04/30/tottenham-vs-afc-ajax',
    'hidden': false,
    'parent': {
      'name': 'Champions League'
    },
    'id': '1943310',
    'inplay_enabled': true,
    'modified': '2019-04-18T10:11:56.786364Z',
    'name': 'Tottenham Hotspur vs. Ajax',
    'parent_id': '929817',
    'short_name': 'TOT vs. AJA',
    'slug': 'tottenham-afc-ajax-2019-04-30',
    'special_rules': null,
    'start_date': '2019-04-30',
    'start_datetime': '2019-04-30T19:00:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-18T10:12:03.638683Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/europe-uefa-champions-league/2019/05/01/fc-barcelona-vs-liverpool-fc',
    'hidden': false,
    'parent': {
      'name': 'Champions League'
    },
    'id': '1943311',
    'inplay_enabled': true,
    'modified': '2019-04-18T10:12:03.638683Z',
    'name': 'Barcelona vs. Liverpool',
    'parent_id': '929817',
    'short_name': 'BAR vs. LIV',
    'slug': 'fc-barcelona-liverpool-fc-2019-05-01',
    'special_rules': null,
    'start_date': '2019-05-01',
    'start_datetime': '2019-05-01T19:00:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-20T10:06:14.610452Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/premier-league/2019/05/04/wolverhampton-vs-fulham-fc',
    'hidden': false,
    'parent': {
      'name': 'Premier League'
    },
    'id': '1958062',
    'inplay_enabled': true,
    'modified': '2019-04-20T10:06:14.610452Z',
    'name': 'Wolverhampton Wanderers vs. Fulham',
    'parent_id': '943086',
    'short_name': 'WOL vs. FUL',
    'slug': 'wolverhampton-fulham-fc-2019-05-04',
    'special_rules': null,
    'start_date': '2019-05-04',
    'start_datetime': '2019-05-04T14:00:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  }
];
