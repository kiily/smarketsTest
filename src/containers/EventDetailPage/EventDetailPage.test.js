import React from 'react';
import { create } from 'react-test-renderer';
import EventDetailPage from './EventDetailPage';
import { BrowserRouter } from 'react-router-dom';
import { events } from '../../mock-data/event-data';

describe('EventDetailPage component', () => {
  it('should match the snapshot when rendered before events load', () => {
    const component = create(<BrowserRouter><EventDetailPage match={{ params: { id: 1925896 } }} popularEvents={events} /></BrowserRouter>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
