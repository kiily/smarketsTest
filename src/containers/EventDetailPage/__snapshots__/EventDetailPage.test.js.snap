// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetailPage component should match the snapshot when rendered before events load 1`] = `null`;
